src/lib/comms/sol-lwm2m.c

```diff
@@ -930,6 +1050,75 @@ static const struct sol_coap_resource registration_interface = {
     }
 };
 
+static int
+bootstrap_request(void *data, struct sol_coap_server *coap,
+    const struct sol_coap_resource *resource,
+    struct sol_coap_packet *req,
+    const struct sol_network_link_addr *cliaddr)
+{
+    struct sol_lwm2m_bootstrap_client_info *bs_cinfo;
+    struct sol_lwm2m_bootstrap_server *server = data;
+    struct sol_coap_packet *response;
+    int r, i;
+    bool know_client = false;
+
+    SOL_DBG("Client Bootstrap Request received");
+
+    response = sol_coap_packet_new(req);
+    SOL_NULL_CHECK(response, -ENOMEM);
+
+    r = new_bootstrap_client_info(&bs_cinfo, cliaddr);
+    SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+
+    r = fill_bootstrap_client_info(bs_cinfo, req);
+    SOL_INT_CHECK_GOTO(r, < 0, err_exit_del_client);
+
+    // TODO: Should I keep clients like this for some reason?
+/*
+    r = sol_ptr_vector_append(&server->clients, cinfo);
+    SOL_INT_CHECK_GOTO(r, < 0, err_exit_del_client);
+ */
+
+    for (i = 0; server->known_clients[i]; i++) {
+        if (strcmp(server->known_clients[i], bs_cinfo->name) == 0)
```

Comment:
Changed the type of `i`.

Regarding the first comment, it should be enough to check if `known_clients` is `NULL` in `sol_lwm2m_bootstrap_server_new()`, right? `known_clients` is supposed to be `NULL`-terminated itself.